### PR TITLE
Add release process

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Documents the release process

Also adds a make rule to build the artifacts

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

/hold
Depends on #195

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Release process
```
Comment (ArangoGutierrez) [inline RELEASE.md:1]: we already have https://github.com/kubernetes-sigs/kueue/blob/main/.github/ISSUE_TEMPLATE/NEW_RELEASE.md , so we should sync this 2 lists

Comment (alculquicondor) [reply RELEASE.md:1]: Done

**Review by ArangoGutierrez — APPROVED** [review]
/lgtm
/hold

File: .github/ISSUE_TEMPLATE/NEW_RELEASE.md [hunk 2]
```diff
@@ -29,19 +30,22 @@ Please do not remove items from the checklist
       `git push $VERSION`
   - Triggers prow to build and publish a staging container image
       `gcr.io/k8s-staging-kueue/kueue:$VERSION`
-- [ ] Submit a PR against [k8s.io](https://github.com/kubernetes/k8s.io), updating `k8s.gcr.io/images/k8s-staging-kueue/images.yaml` to promote the container images (both "full" and "minimal" variants) to production
-- [ ] Wait for the PR to be merged and verify that the image (`k8s.gcr.io/kueue/kueue:$VERSION`) is available.
-- [ ] Publish the draft release prepared at the [Github releases page](https://github.com/kubernetes-sigs/kueue/releases)
-      which will also trigger a Helm repo index update to add the latest release
+- [ ] An OWNER runs
+      `make artifacts IMAGE_REGISTRY=k8s.gcr.io/kueue GIT_TAG=$VERSION`
+      to generate the artifacts and uploads the files in the `artifacts` folder
```
[the remark this answers is not in the document]
Comment (alculquicondor) [reply .github/ISSUE_TEMPLATE/NEW_RELEASE.md]: Moved this line up and removed old line 25.

File: .github/ISSUE_TEMPLATE/NEW_RELEASE.md [hunk 3]
```diff
@@ -29,19 +30,22 @@ Please do not remove items from the checklist
...
+      to the release.
```
[the remark this answers is not in the document]
Comment (alculquicondor) [reply .github/ISSUE_TEMPLATE/NEW_RELEASE.md]: I was referring to the draft in the release tool. Changed.

**Pull request merged**